src/main.rs

```diff
@@ -211,6 +216,18 @@ fn main2() -> Result<(), Box<dyn Error>> {
         let holdout_after_option: Option<u64> =
             cl.value_of("holdout_after").map(|s| s.parse().unwrap());
 
+        
+        let hogwild_training = cl.is_present("hogwild_training");
+        let mut hogwild_trainer = if hogwild_training {
+            let hogwild_threads = match cl.value_of("hogwild_threads") {
+                Some(hogwild_threads) => hogwild_threads.parse().expect("hogwild_threads should be integer"),
+                None => 16
```

Comment:
Is 16 a bit much for a default?

Reply:
Ideally this would be dynamic according to the system's cores... but couldn't get it to work without adding a new crate. regarding the correct number - not sure yet. Will leave it as is for.